Microsoft.HealthVault.Fhir/Codings/CodeToHealthVaultHelper.cs

```diff
@@ -107,5 +111,53 @@ private static Type DetectFromHealthVaultCode(string code)
 
             throw new NotSupportedException("The provided code is not supported");
         }
+
+        internal static Type DetectHealthVaultTypeFromDocumentReference(DocumentReference documentReference)
+        {
+            var contentComponent = documentReference.GetFirstContentComponentWithData();
+
+            if (contentComponent == null)
+            {
+                throw new ArgumentException("DocumentReference must have a content component with data");
+            }
+
+            if (contentComponent.Attachment.ContentType == "text/xml"
+                || contentComponent.Attachment.ContentType == "application/xml")
+            {
+                string xml = Encoding.UTF8.GetString(contentComponent.Attachment.Data);
+
+                XPathDocument xpDoc;
+                try
+                {
+                    using (System.IO.TextReader txtReader = new System.IO.StringReader(xml))
+                    {
+                        xpDoc = new XPathDocument(txtReader);
+                    }
+
+                    XPathNavigator xpNav = xpDoc.CreateNavigator();
+                    xpNav.MoveToFirstChild();
+
+                    do
+                    {
+                        if (xpNav.NodeType == XPathNodeType.Element)
+                        {
+                            switch (xpNav.Name)
+                            {
+                                case "ContinuityOfCareRecord":
+                                    return typeof(CCR);
+                                case "ClinicalDocument":
+                                    return typeof(CDA);
+                            }
+                        }
+                    } while (xpNav.MoveToNext());
+                }
+                catch (XmlException)
```

Comment:
What's the thinking behind returning the type of "File" if we run into an xml exception (xml sytax error)?

Reply:
The idea was that when the xml is found to be invalid, the conversion defaults to that of a file as it  can no longer be considered a clinical document.